x/incentives/keeper/gauge_test.go

```diff
@@ -280,33 +114,92 @@ func (suite *KeeperTestSuite) TestGaugesByDenom() {
+		suite.Require().Len(gaugeIdsByDenom, 0)
 
 		// distribute coins to stakers
-		_, err = suite.App.IncentivesKeeper.Distribute(suite.Ctx, []types.Gauge{*gauge})
+		distrCoins, err := suite.App.IncentivesKeeper.Distribute(suite.Ctx, []types.Gauge{*gauge})
 		suite.Require().NoError(err)
+		suite.Require().Equal(sdk.Coins{sdk.NewInt64Coin("stake", int64(10/tc.expectedNumEpochsPaidOver))}, distrCoins)
+
+		if tc.isPerpetual {
+			// distributing twice without adding more for perpetual gauge
+			gauge, err = suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, gaugeID)
+			suite.Require().NoError(err)
+			distrCoins, err = suite.App.IncentivesKeeper.Distribute(suite.Ctx, []types.Gauge{*gauge})
+			suite.Require().NoError(err)
+			suite.Require().True(distrCoins.Empty())
+
+			// add to gauge
+			addCoins := sdk.Coins{sdk.NewInt64Coin("stake", 200)}
+			suite.AddToGauge(addCoins, gaugeID)
+
+			// distributing twice with adding more for perpetual gauge
+			gauge, err = suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, gaugeID)
+			suite.Require().NoError(err)
+			distrCoins, err = suite.App.IncentivesKeeper.Distribute(suite.Ctx, []types.Gauge{*gauge})
+			suite.Require().NoError(err)
+			suite.Require().Equal(sdk.Coins{sdk.NewInt64Coin("stake", 200)}, distrCoins)
+		}
+
+		// check active gauges
+		gauges = suite.App.IncentivesKeeper.GetActiveGauges(suite.Ctx)
+		suite.Require().Len(gauges, 1)
+
+		// check gauge ids by denom
+		gaugeIdsByDenom = suite.App.IncentivesKeeper.GetAllGaugeIDsByDenom(suite.Ctx, "lptoken")
+		suite.Require().Len(gaugeIdsByDenom, 1)
 
 		// finish distribution for non perpetual gauge
 		if !gauge.IsPerpetual {
 			err = suite.App.IncentivesKeeper.MoveActiveGaugeToFinishedGauge(suite.Ctx, *gauge)
 			suite.Require().NoError(err)
 		}
 
+		// check finished gauges
+		if !tc.isPerpetual {
+
+			// check finished gauges
+			gauges = suite.App.IncentivesKeeper.GetFinishedGauges(suite.Ctx)
+			suite.Require().Len(gauges, 1)
+
+			// check gauge by ID
+			gauge, err = suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, gaugeID)
+			suite.Require().NoError(err)
+			suite.Require().NotNil(gauge)
+			suite.Require().Equal(gauges[0], *gauge)
+
+			// check invalid gauge ID
+			_, err = suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, gaugeID+1000)
+			suite.Require().Error(err)
+			rewardsEst = suite.App.IncentivesKeeper.GetRewardsEst(suite.Ctx, lockOwner, []lockuptypes.PeriodLock{}, 100)
+			suite.Require().Equal(sdk.Coins{}, rewardsEst)
+		} else {
+			// check finished gauges
+			gauges = suite.App.IncentivesKeeper.GetFinishedGauges(suite.Ctx)
+			suite.Require().Len(gauges, 0)
+
+			// check rewards estimation
+			rewardsEst = suite.App.IncentivesKeeper.GetRewardsEst(suite.Ctx, lockOwner, []lockuptypes.PeriodLock{}, 100)
+			suite.Require().Equal(sdk.Coins(nil), rewardsEst)
+		}
+
 		expectedNumGauges := 1
-		if !isPerpetual {
+		if !tc.isPerpetual {
```

Comment:
Since we now have separate perpetual/non-perpetual blocks, why not just move these instances of `expectedNumGauges` to the end of each of those?

e.g. put `expectedNumGauges := 1` at the end of the `if tc.isPerpetual {...}` section and `expectedNumGauges := 0` at the end of the `if !tc.isPerpetual {...}` section

This would also have the added benefit of reducing our # of if statements to two, one for perp gauges and one for non-perp, which imo improves code readability a lot

Reply:
Makes sense -- the additional if statement just adds more confusion.

I made the change (putting the declaration outside the if/else blocks because of variable scope), but it might just be cleaner to put the denom checks directly at the end of both blocks? Lmk if you feel strongly either way

Reply:
Denom checks at the end of both blocks would be the most appropriate in this case imo